### PR TITLE
LPS-144248 Avoid DoS attack

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java [hunk 1]
```diff
@@ -519,9 +523,29 @@ protected void setDisplayContext(
...
+				"name" + LocaleUtil.toLanguageId(availableLocale));
+		}
+
+		return localizedParameters;
+	}
+
 	private static final Log _log = LogFactoryUtil.getLog(
 		KaleoFormsAdminPortlet.class);
 
+	private static final List<String> _regularParameters = Arrays.asList(
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:542]: @MarinhoFeliphe rename to `_parameterNames`, please.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java [hunk 2]
```diff
@@ -519,9 +523,29 @@ protected void setDisplayContext(
 	@Reference
 	protected StorageEngine storageEngine;
 
+	private List<String> _getLocalizedParameters() {
+		List<String> localizedParameters = new ArrayList<>();
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:527]: @MarinhoFeliphe rename to `_getLocalizedParameterNames`, please.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java [hunk 3]
```diff
@@ -377,26 +383,24 @@ protected void saveInPortletSession(
...
-		Enumeration<String> enumeration = resourceRequest.getParameterNames();
+		Map<String, String[]> parameterMap = resourceRequest.getParameterMap();
 
-		while (enumeration.hasMoreElements()) {
-			String name = enumeration.nextElement();
+		PortletSession portletSession = resourceRequest.getPortletSession();
 
-			if (Objects.equals(name, "doAsUserId")) {
+		for (String regularParameter :
+				ListUtil.concat(
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:395]: @MarinhoFeliphe rename to parameterName, please.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java [hunk 4]
```diff
@@ -377,26 +383,24 @@ protected void saveInPortletSession(
...
+					_getLocalizedParameters(), _regularParameters)) {
+
+			if (parameterMap.get(regularParameter) == null) {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:398]: @MarinhoFeliphe use `parameterMap.containsKey(regularParameter)` instead

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java (hunk 2 above)
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:527]: @MarinhoFeliphe rename to `localizedParameterNames`, please.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:help

**Comment by MarinhoFeliphe** [comment]
ci:help

**Comment by brianchandotcom** [comment]
@MarinhoFeliphe thanks for the fix! See my changes in 566ba7b48d6e8c62e5da71c34bb56b87183bf503

**Comment by MarinhoFeliphe** [comment]
ci:test:kaleo-forms

**Pull request closed without merging**